app/src/main/kotlin/com/saltedge/authenticator/features/consents/ConsentsListViewModel.kt

```diff
@@ -0,0 +1,179 @@
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * For the additional permissions granted for Salt Edge Authenticator
+ * under Section 7 of the GNU General Public License see THIRD_PARTY_NOTICES.md
+ */
+package com.saltedge.authenticator.features.consents
+
+import android.content.Context
+import androidx.lifecycle.*
+import com.saltedge.authenticator.R
+import com.saltedge.authenticator.features.authorizations.common.collectConnectionsAndKeys
+import com.saltedge.authenticator.features.connections.common.ConnectionViewModel
+import com.saltedge.authenticator.features.connections.list.collectConnectionViewModel
+import com.saltedge.authenticator.features.consents.common.ConsentItemViewModel
+import com.saltedge.authenticator.models.Connection
+import com.saltedge.authenticator.models.ViewModelEvent
+import com.saltedge.authenticator.models.repository.ConnectionsRepositoryAbs
+import com.saltedge.authenticator.sdk.AuthenticatorApiManagerAbs
+import com.saltedge.authenticator.sdk.contract.FetchEncryptedDataListener
+import com.saltedge.authenticator.sdk.model.ConnectionID
+import com.saltedge.authenticator.sdk.model.ConsentData
+import com.saltedge.authenticator.sdk.model.EncryptedData
+import com.saltedge.authenticator.sdk.model.connection.ConnectionAndKey
+import com.saltedge.authenticator.sdk.model.error.ApiErrorData
+import com.saltedge.authenticator.sdk.tools.crypt.CryptoToolsAbs
+import com.saltedge.authenticator.sdk.tools.keystore.KeyStoreManagerAbs
+import com.saltedge.authenticator.tools.daysTillExpire
+import kotlinx.coroutines.*
+import kotlin.coroutines.CoroutineContext
+
+class ConsentsListViewModel(
+    private val appContext: Context,
+    private val connectionsRepository: ConnectionsRepositoryAbs,
+    private val keyStoreManager: KeyStoreManagerAbs,
+    private val apiManager: AuthenticatorApiManagerAbs,
+    private val cryptoTools: CryptoToolsAbs
+) : ViewModel(), LifecycleObserver, FetchEncryptedDataListener, CoroutineScope {
+
+    private val decryptJob: Job = Job()
+    override val coroutineContext: CoroutineContext
+        get() = decryptJob + Dispatchers.IO
+
+    val listItems = MutableLiveData<List<ConsentItemViewModel>>()
+    val connectionItem = MutableLiveData<ConnectionViewModel>()
+
+    private var connection = Connection()
+    private var connectionGuid = ""
+    private var connectionsAndKeys: Map<ConnectionID, ConnectionAndKey> =
+        collectConnectionsAndKeys(
+            connectionsRepository,
+            keyStoreManager
+        )
+    var onListItemClickEvent = MutableLiveData<ViewModelEvent<Int>>()
```

Comment:
pass through `onListItemClickEvent` Bundle with `connection_guid` and `ConsentData`